### PR TITLE
Few updates to the UI contributing guide

### LINKED ISSUE #2304: [Docs] Meshery UI Contributor Guide
**Current State:**
Meshery UI has a number of aspects to it like use of rest, graphql, nextjs, react, billboardjs and so on.

New and existing contributors to Meshery's web user interface will benefit from a contributor guide.

**Desired State:**
- Create a README.md in the /ui folder that serves as a contributor guide to this component.
- Add link to this README in the main Meshery CONTRIBUTING.md.
- Include description of things like:
   - Meshery UI doing server-side rendering of NextJS/React components.
   - Meshery UI uses billboard.js as the charting library. 

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Signed-off-by: Nikhil-Ladha <nikhilladha1999@gmail.com>

This PR fixes #2304 

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: ui/README.md [hunk 1]
```diff
@@ -1,7 +1,33 @@
 # Contributor Guide for UI component
 
-<p style="text-align: center"><em>If you'll like to go to the main Meshery Contributor guide <a href="../CONTRIBUTING.md">click here</a></em></p>
+This guide is specific to the Meshery UI component and involves steps/methods one need to follow while working on issues related to Meshery UI.
 
-## Meshery UI doing server-side rendering of NextJS/React components.
+## How to run Meshery UI?
+Meshery UI can be used for different purposes, it can be for general use like testing out Meshery and it's functionalities or work on issues related to the Meshery UI.
```
Comment (leecalcote) [inline ui/README.md:6]:
```suggestion
Meshery UI can be built and run in different ways. You will choose one of the two ways to build and run Meshery UI depending upon whether you are actively developing it (whether you are creating a new feature or fixing a bug in Meshery UI) or whether you simply need to use it as a user. Let's refer to these two methods as a _Development Build_ and _User Build._
```

File: ui/README.md [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
 
-## Meshery UI uses billboard.js as the charting library.
+#### 1. General Usage:
```
Comment (leecalcote) [inline ui/README.md:8]:
```suggestion
#### 1. User Build:
```

File: ui/README.md [hunk 3]
```diff
@@ -1,7 +1,33 @@
...
+For general usage, one can run Meshery UI using Meshery's command client `mesheryctl`, by simply running the `mesheryctl system start` command.
+If you don't have the `mesheryctl` tool installed already, you can follow the [mesheryctl installation docs](https://docs.meshery.io/installation/mesheryctl) to install `mesheryctl` using various `package management` tools supported.
+
+#### 2. For development purporse:
```
Comment (leecalcote) [inline ui/README.md:12]:
```suggestion
#### 2. Development Build:
```

File: ui/README.md [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
+For development purpose, either of the below mentioned approach could be followed
```
Comment (leecalcote) [inline ui/README.md:13]:
```suggestion
For purposes of actively developing Meshery UI, you can use either of the following approaches.
```

File: ui/README.md [hunk 5]
```diff
@@ -1,7 +1,33 @@
...
+- Follow the procedure mentioned in Step 1 (General Usage) above, and start Meshery UI sever on the 9081 port, and login to Meshery UI using eihter of the providers mentioned on the login page. Then, to run a development server of Meshery UI, first install the dependencies using the `make setup-ui-libs` command, then build the UI package and export it using the command `make build-ui`, and finally execute `make run-ui-dev` to run the livereload-nodemon server on port 3000.
+    > **NOTE:** Please run the steps in order to avoid issues, as Meshery server should be running and logged-in before accessing the development server
+    > on 3000 port.
+    
+- Another way, to run Meshery UI for development purpose is by using the `make run-local` command. But, before that you will have to build the Meshery UI and export it using `make buil-ui` command. This method doesn't give a livereload server and one will have to build the Meshery-UI after making changes to the code and then again run the server.
```
Comment (leecalcote) [inline ui/README.md:18]:
```suggestion
2)  **`make run-local`** - Alternatively, build all of Meshery UI's components upfront before serving the UI. Do this in two steps:
# Execute `make build-ui` to build and export all Meshery UI components.
# Execute `make run-local` to serve the prebuilt components. 
This method doesn't provide a live reload server. You will have to build Meshery UI after making changes to the code and rerun these steps again in order to see those subsequent code changes reflected in the UI.
```

File: ui/README.md [hunk 6]
```diff
@@ -1,7 +1,33 @@
...
 
-## Meshery UI uses billboard.js as the charting library.
+#### 1. General Usage:
+For general usage, one can run Meshery UI using Meshery's command client `mesheryctl`, by simply running the `mesheryctl system start` command.
+If you don't have the `mesheryctl` tool installed already, you can follow the [mesheryctl installation docs](https://docs.meshery.io/installation/mesheryctl) to install `mesheryctl` using various `package management` tools supported.
+
+#### 2. For development purporse:
+For development purpose, either of the below mentioned approach could be followed
+- Follow the procedure mentioned in Step 1 (General Usage) above, and start Meshery UI sever on the 9081 port, and login to Meshery UI using eihter of the providers mentioned on the login page. Then, to run a development server of Meshery UI, first install the dependencies using the `make setup-ui-libs` command, then build the UI package and export it using the command `make build-ui`, and finally execute `make run-ui-dev` to run the livereload-nodemon server on port 3000.
```
Comment (leecalcote) [inline ui/README.md:14]: @Nikhil-Ladha, does `make-setup-libs` need to be pulled out and identified as a one-time prerequisite command to execute irrespective of which method the developer uses?

Comment (Nikhil-Ladha) [reply ui/README.md:14]: I think so, yes.

File: ui/README.md [hunk 7]
```diff
@@ -1,7 +1,36 @@
...
+
+## How to run Meshery UI?
+Meshery UI can be built and run in different ways. You will choose one of the two ways to build and run Meshery UI depending upon whether you are actively developing it (whether you are creating a new feature or fixing a bug in Meshery UI) or whether you simply need to use it as a user. Let's refer to these two methods as a _Development Build_ and _User Build._
+
+#### 1. User Build:
+For general usage, one can run Meshery UI using Meshery's command client `mesheryctl`, by simply running the `mesheryctl system start` command.
+If you don't have the `mesheryctl` tool installed already, you can follow the [mesheryctl installation docs](https://docs.meshery.io/installation/mesheryctl) to install `mesheryctl` using various `package management` tools supported.
+
+#### 2. Development Build:
+For purposes of actively developing Meshery UI, you first need to install the dependeincies using `make setup-ui-libs` and then you can use either of the following approaches to build Meshery UI:
```
Comment (subhamkrai) [inline ui/README.md:13]:
```suggestion
For purposes of actively developing Meshery UI, you first need to install the dependencies using `make setup-ui-libs` and then you can use either of the following approaches to build Meshery UI:
```

File: ui/README.md [hunk 8]
```diff
@@ -1,7 +1,36 @@
...
+1. Follow the procedure mentioned in Step 1 (User build) above, and start Meshery UI sever on the 9081 port, and login to Meshery UI using eihter of the providers mentioned on the login page. Then, to run a development server of Meshery UI, install the dependencies using the command mentioned above, then build the UI package and export it using the command `make build-ui`, and finally execute `make run-ui-dev` to run the livereload-nodemon server on port 3000.
```
Comment (subhamkrai) [inline ui/README.md:14]:
```suggestion
1. Follow the procedure mentioned in Step 1 (User build) above, and start Meshery UI sever on the 9081 port, and login to Meshery UI using either of the providers mentioned on the login page. Then, to run a development server of Meshery UI, install the dependencies using the command mentioned above, then build the UI package and export it using the command `make build-ui`, and finally execute `make run-ui-dev` to run the livereload-nodemon server on port 3000.
```

File: ui/README.md [hunk 9]
```diff
@@ -1,7 +1,36 @@
...
+- Meshery UI uses NextJs to do server side rendering of ReactJS components. The folder `ui/components` contains all the ReactJS components involved in
+  building Meshery UI.
+- MaterialUI is being used extensively for the visual components of Meshery UI.
+- Billboard.js library is being used to display various charts, and comparison graphs in Meshery UI.
 
-## Meshery UI uses billboard.js as the charting library.
+
+## Component naming convention
+For reference and easy code search, the components are named accordingly following the rule 'Meshery<Part of UI it involves>', for example: components
+involved in rednering the Results page of Meshery UI are named as 'MesheryResults.js', 'MesheryResultDialog.js', 'MesherySMIResults.js'. Please follow this convention if you are creating a new component.
```
Comment (subhamkrai) [inline ui/README.md:33]:
```suggestion
involved in rendering the Results page of Meshery UI are named as 'MesheryResults.js', 'MesheryResultDialog.js', 'MesherySMIResults.js'. Please follow this convention if you are creating a new component.
```

File: ui/README.md [hunk 10]
```diff
@@ -1,7 +1,36 @@
 # Contributor Guide for UI component
 
-<p style="text-align: center"><em>If you'll like to go to the main Meshery Contributor guide <a href="../CONTRIBUTING.md">click here</a></em></p>
+This guide is specific to the Meshery UI component and involves steps/methods one need to follow while working on issues related to Meshery UI.
```
Comment (subhamkrai) [inline ui/README.md:3]:
```suggestion
This guide is specific to the Meshery UI component and involves steps/methods one needs to follow while working on issues related to Meshery UI.
```

**Review by subhamkrai — CHANGES_REQUESTED** [review]

File: ui/README.md [hunk 11]
```diff
@@ -1,7 +1,36 @@
...
+  building Meshery UI.
+- MaterialUI is being used extensively for the visual components of Meshery UI.
+- Billboard.js library is being used to display various charts, and comparison graphs in Meshery UI.
 
-## Meshery UI uses billboard.js as the charting library.
+
+## Component naming convention
+For reference and easy code search, the components are named accordingly following the rule 'Meshery<Part of UI it involves>', for example: components
+involved in rednering the Results page of Meshery UI are named as 'MesheryResults.js', 'MesheryResultDialog.js', 'MesherySMIResults.js'. Please follow this convention if you are creating a new component.
+
```
Comment (subhamkrai) [inline ui/README.md:34]: remove extra empty line
```suggestion
```

File: ui/README.md (hunk 10 above)
Comment (Nikhil-Ladha) [reply ui/README.md:3]: Not needed.

File: ui/README.md (hunk 11 above)
Comment (Nikhil-Ladha) [reply ui/README.md:34]: Intentional, not needed.

**Review by leecalcote — APPROVED** [review]
Great update.

**Review by jomefavourite — APPROVED** [review]

**Review by jomefavourite — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**